src/Phinx/Migration/Manager.php

```diff
@@ -847,6 +847,24 @@ function ($phpFile) {
         return $this->migrations;
     }
 
+    /**
+     * Returns a list of the versions of the migration file
+     * array_keys() of $this->migrations is not reliable on 32 bit systems
+     *
+     * @param string $environment Environment
+     * @return string[]
+     */
+    public function getVersions(string $environment): array
```

Comment:
To clarify, this new function isn't necessary for the rest of the PR changes, just something you'd like downstream?

As a note to myself, on merging this PR, will want to manually do it such that we have two commits, one that does all the other changes and then a second that has just this new function.

Reply:
This will be required by cakephp/migrations. We cannot rely on the array_keys of the migtations.
See here: https://github.com/passbolt/migrations/commit/46a3e7bf6f26e71b7c4287497b6d2e47eded1ae2#diff-e60533c90c4ac309c99359e07ec316f9e774c60cacf1ad2d7dfdd615a17a96ebR129

So this new method is needed, and IMO part of the casting to string.

Reply:
Considering your comment above, this might be handled differently though.